src/utils/SparseGrid.h

```diff
@@ -27,8 +27,10 @@ class SparseGridInvasive
      *
      * \param[in] cell_size The size to use for a cell (square) in the grid.
      *    Typical values would be around 0.5-2x of expected query radius.
+     * \param[in] elem_reserve Number of elements to research space for.
+     * \param[in] max_load_factor Maximum average load factor before rehashing.
      */
-    SparseGridInvasive(coord_t cell_size);
+    SparseGridInvasive(coord_t cell_size, size_t elem_reserve=0U, float max_load_factor=1.0f);
```

Comment:
It could still be potentially useful so I left it in.  If because of the input data you are expecting to have pairs of points near each other, it might be useful to set the load factor as 2.0 as otherwise about half the buckets will be empty.  At least if I understand the unordered_map implementation well enough.